converters/toZigbee.js

```diff
@@ -4849,6 +4849,60 @@ const converters = {
             await entity.read('schneiderSpecificPilotMode', ['pilotMode'], {manufacturerCode: 0x105e});
         },
     },
+    schneider_temperature_measured_value: {
+        key: ['temperature_measured_value'],
+        convertSet: async (entity, key, value, meta) => {
+            await entity.report('msTemperatureMeasurement', {'measuredValue':Math.round(value*100)});
+        },
+    },
+    schneider_thermostat_system_mode: {
+        key: ['system_mode'],
+        convertSet: async (entity, key, value, meta) => {
+            let systemMode = utils.getKey(constants.thermostatSystemModes, value, undefined, Number);
+            entity.saveClusterAttributeKeyValue('hvacThermostat', {systemMode: systemMode});
+            return {state: {system_mode: value}};
+        },
+    },
+    schneider_thermostat_occupied_heating_setpoint: {
+        key: ['occupied_heating_setpoint'],
+        convertSet: async (entity, key, value, meta) => {
+            const occupiedHeatingSetpoint = (Math.round((value * 2).toFixed(1)) / 2).toFixed(1) * 100;
+            entity.saveClusterAttributeKeyValue('hvacThermostat', {occupiedHeatingSetpoint: occupiedHeatingSetpoint});
+            return {state: {occupied_heating_setpoint: value}};
+        },
+    },
+    schneider_thermostat_control_sequence_of_operation: {
+        key: ['control_sequence_of_operation'],
+        convertSet: async (entity, key, value, meta) => {
+            let val = utils.getKey(constants.thermostatControlSequenceOfOperations, value, undefined, Number);
+            if (val === undefined) {
+                val = utils.getKey(constants.thermostatControlSequenceOfOperations, value, value, Number);
+            }
+            entity.saveClusterAttributeKeyValue('hvacThermostat', {ctrlSeqeOfOper: val});
+            return {state: {control_sequence_of_operation: value}};
+        },
+    },
+    schneider_thermostat_pi_heating_demand: {
+        key: ['pi_heating_demand'],
+        convertSet: async (entity, key, value, meta) => {
+            entity.saveClusterAttributeKeyValue('hvacThermostat', {pIHeatingDemand: value});
+            return {state: {pi_heating_demand: value}};
+        },
+    },
+    schneider_thermostat_keypad_lockout: {
+        key: ['keypad_lockout'],
+        convertSet: async (entity, key, value, meta) => {
+            const keypadLockout = utils.getKey(constants.keypadLockoutMode, value, value, Number);
+
+            try {
+                await entity.write('hvacUserInterfaceCfg', {keypadLockout}, {timeout: 500});
+            } catch(e) {
+                // Will probably fail if device is not awake
+                globalStore.putValue(entity, 'pending_writes', {cluster: 'hvacUserInterfaceCfg', data: keypadLockout.keypadLockout},);
```

Comment:
Instead of the `pending_writes` thing I've added a new option: `sendWhenActive`. This will send the message when the coordinator receives a message from it. Can you test it? You can remove this whole try catch block and replace it with: `await entity.write('hvacUserInterfaceCfg', {keypadLockout}, {timeout: 500, sendWhenActive: true});` 

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html)

Reply:
I tested it and something seems not right: on the device the "lock" indication is spuriously toggling between on and off.
Also, the readback of "saved cluster attributes" seems unstable.

Do you have any idea of what could be wrong? 
I'll try to investigate a bit deeper tomorow.

Reply:
You can look at the herdsman debug log to see what is happening.

See https://www.zigbee2mqtt.io/information/debug.html#zigbee-herdsman-debug-logging on how to enable the herdsman debug logging. Note that this is only logged to STDOUT and not to log files.

Reply:
Sorry I did not comment back, but this is working fine after the pull request on herdsman you just closed.